lib/NonlinearSolveBase/src/solve.jl

```diff
@@ -1,3 +1,449 @@
+const allowedkeywords = (:dense,
```

Comment:
The utilities at the top here are reused by DiffEqBase, NonlinearSolveBase, OptimizationBase (in the near future), etc. Let's move these to SciMLBase and share them across the different repos.

Reply:
The list of acceptable kwargs, yes that one should be problem specific, that would be a nice improvement.

But the others yes.